frontend/app/src/scenes/Home/scenes/AddSubCategory/index.tsx

```diff
@@ -102,25 +146,61 @@ class AddSubCategory extends React.Component<
                     <Text>Select a category</Text>
-                    <Form.Item name="category" className={mainStyles.formItem}>
-                      <Select
-                        className={mainStyles.formSelect}
-                        placeholder="Please select a category"
+                    {this.state.subcategory != null && (
+                      <Form.Item
+                        name="category"
+                        className={mainStyles.formItem}
+                        initialValue={this.state.subcategory.category.id}
                       >
-                        {this.state.categories.map((category: Category) => {
-                          return (
-                            <Option key={category.id} value={category.id}>
-                              {category.name}
-                            </Option>
-                          );
-                        })}
-                      </Select>
-                    </Form.Item>
+                        <Select
+                          className={mainStyles.formSelect}
+                          placeholder="Please select a category"
+                        >
+                          {this.state.categories.map((category: Category) => {
+                            return (
+                              <Option key={category.id} value={category.id}>
+                                {category.name}
+                              </Option>
+                            );
+                          })}
+                        </Select>
+                      </Form.Item>
+                    )}
+                    {this.state.subcategory == null && (
+                      <Form.Item
+                        name="category"
+                        className={mainStyles.formItem}
+                      >
+                        <Select
+                          className={mainStyles.formSelect}
+                          placeholder="Please select a category"
+                        >
+                          {this.state.categories.map((category: Category) => {
+                            return (
+                              <Option key={category.id} value={category.id}>
+                                {category.name}
+                              </Option>
+                            );
+                          })}
+                        </Select>
+                      </Form.Item>
+                    )}
                     <Title level={3}>Name</Title>
                     <Text>Name of the subcategory</Text>
-                    <Form.Item name="name" className={mainStyles.formItem}>
-                      <Input placeholder="ex: Local Curriculum" />
-                    </Form.Item>
+                    {this.state.subcategory != null && (
+                      <Form.Item
+                        name="name"
+                        className={mainStyles.formItem}
+                        initialValue={this.state.subcategory.name}
```

Comment:
pass the initial value object (subcategory) to the form. not to the form items

Reply:
ok